src/modules/app/StoreActionType.js

```diff
@@ -1,180 +1,224 @@
+import type { ContentLoadCriterionType } from '../endpoint/ContentLoadCriterion'
+import type { TFunction } from 'react-i18next'
+import type { ErrorCodeType } from '../error/ErrorCodes'
+
+export type FetchCitiesActionType = {|
+  type: 'FETCH_CITIES',
+  +params: {| +forceRefresh: boolean |}
+|}
+export type PushCitiesActionType = {|
+  type: 'PUSH_CITIES', +params: {| +cities: $ReadOnlyArray<CityModel> |}
+|}
+export type FetchCitiesFailedActionType = {|
+  type: 'FETCH_CITIES_FAILED', +params: {|
+    +message: string, +code: ErrorCodeType
+  |}
+|}
+export type CitiesActionType = PushCitiesActionType | FetchCitiesActionType | FetchCitiesFailedActionType
+
+export type PushLanguagesActionType = {|
+  type: 'PUSH_LANGUAGES',
+  +params: {|
+    +languages: $ReadOnlyArray<LanguageModel>
+  |}
+|}
+export type FetchLanguagesFailedActionType = {|
+  type: 'FETCH_LANGUAGES_FAILED',
+  +params: {|
+    +message: string,
+    +code: ErrorCodeType
+  |}
+|}
+
+export type SetContentLanguageActionType = {|
+  type: 'SET_CONTENT_LANGUAGE', +params: {| +contentLanguage: string |}
+|}
+
+export type FetchCategoryActionType = {|
+  type: 'FETCH_CATEGORY',
+  +params: {|
+    +key: string,
+    ...CategoryRouteConfigType,
+    +criterion: ContentLoadCriterionType
+  |}
+|}
+export type FetchCategoryFailedActionType = {|
+  type: 'FETCH_CATEGORY_FAILED',
+  +params: {|
+    +key: string,
+    ...CategoryRouteConfigType,
+    +message: string,
+    +code: ErrorCodeType,
+    +allAvailableLanguages: $ReadOnlyMap<string, ?string> | null
+  |}
+|}
+export type PushCategoryActionType = {|
+  type: 'PUSH_CATEGORY',
+  +params: {|
+    +categoriesMap: CategoriesMapModel,
+    +resourceCache: LanguageResourceCacheStateType,
+    +cityLanguages: Array<LanguageModel>,
+    ...CategoryRouteConfigType,
+    +key: string
+  |}
+|}
+
+export type ClearCategoryActionType = {|
+  type: 'CLEAR_CATEGORY', +params: {| +key: string |}
+|}
+
+export type CategoriesActionType =
+  ClearCategoryActionType
+  | FetchCategoryActionType
+  | PushCategoryActionType
+  | FetchCategoryFailedActionType
+
+export type FetchPoiActionType = {|
+  type: 'FETCH_POI', +params: {|
+    +city: string, +language: string,
+    +path: ?string, +key: string,
+    +criterion: ContentLoadCriterionType
+  |}
+|}
+
+export type ClearPoiActionType = {|
+  type: 'CLEAR_POI', +params: {| +key: string |}
+|}
+
+export type PushPoiActionType = {|
+  type: 'PUSH_POI',
+  +params: {|
+    +pois: $ReadOnlyArray<PoiModel>,
+    +path: ?string,
+    +key: string,
+    +resourceCache: LanguageResourceCacheStateType,
+    +cityLanguages: $ReadOnlyArray<LanguageModel>,
+    +language: string,
+    +city: string
+  |}
+|}
+export type FetchPoiFailedActionType = {|
+  type: 'FETCH_POI_FAILED',
+  +params: {|
+    +message: string,
+    +code: ErrorCodeType,
+    +key: string,
+    +allAvailableLanguages: ?$ReadOnlyMap<string, ?string>,

```

Comment:
```suggestion
    +allAvailableLanguages: ?$ReadOnlyMap<string, string>,

```

This should probably be `string`